teloc/src/service_provider.rs

```diff
@@ -72,70 +74,55 @@ impl Default for ServiceProvider<EmptyServiceProvider, HNil> {
         }
     }
 
     /// Forking `ServiceProvider` creates a new `ServiceProvider` with reference to the parent.
     /// `resolve` method on forked `ServiceProvider` will find dependencies form self and parent.
-    pub fn fork_rc(self: &Rc<ServiceProvider<Parent, Deps>>) -> ServiceProvider<Rc<Self>, HNil> {
+    pub fn fork_rc(self: &Rc<ServiceProvider<Parent, Conts>>) -> ServiceProvider<Rc<Self>, HNil> {
         ServiceProvider {
             parent: self.clone(),
-            dependencies: HNil,
+            containers: HNil,
         }
     }
 
     /// Forking `ServiceProvider` creates a new `ServiceProvider` with reference to the parent.
     /// `resolve` method on forked `ServiceProvider` will find dependencies form self and parent.
-    pub fn fork_arc(self: &Arc<ServiceProvider<Parent, Deps>>) -> ServiceProvider<Arc<Self>, HNil> {
+    pub fn fork_arc(
+        self: &Arc<ServiceProvider<Parent, Conts>>,
+    ) -> ServiceProvider<Arc<Self>, HNil> {
         ServiceProvider {
             parent: self.clone(),
-            dependencies: HNil,
+            containers: HNil,
         }
     }
 }
 
 // Clippy requires to create type aliases
-type ContainerTransientAddConvert<P, T, U, H> =
-    ServiceProvider<P, HCons<ConvertContainer<TransientContainer<T>, T, U>, H>>;
-type ContainerSingletonAddConvert<P, T, U, H> =
-    ServiceProvider<P, HCons<ConvertContainer<SingletonContainer<T>, T, U>, H>>;
-type ContainerInstanceAddConvert<P, T, U, H> =
-    ServiceProvider<P, HCons<ConvertContainer<InstanceContainer<T>, T, U>, H>>;
+type ContainerTransientAddConvert<Parent, T, U, Conts> =
+    ServiceProvider<Parent, HCons<ConvertContainer<TransientContainer<T>, T, U>, Conts>>;
+type ContainerSingletonAddConvert<Parent, T, U, Conts> =
+    ServiceProvider<Parent, HCons<ConvertContainer<SingletonContainer<T>, T, U>, Conts>>;
+type ContainerInstanceAddConvert<Parent, T, U, Conts> =
+    ServiceProvider<Parent, HCons<ConvertContainer<InstanceContainer<T>, T, U>, Conts>>;
 
-impl<Parent, Deps: HList> ServiceProvider<Parent, Deps> {
-    /// Method used primary for internal actions. In common usage you don't need to use it. It add dependencies to the store. You need
-    /// to put in first generic parameter some `ContainerElem` type.
-    /// Usage:
-    ///
-    /// ```
-    /// use teloc::*;
-    /// use teloc::container::TransientContainer;
-    ///
-    /// struct Service {
-    ///     data: i32,
-    /// }
-    ///
-    /// let sp = ServiceProvider::new()
-    ///     ._add::<TransientContainer<Service>>(());
-    /// ```
-    pub fn _add<Container: Init>(
-        self,
-        data: Container::Data,
-    ) -> ServiceProvider<Parent, HCons<Container, Deps>> {
-        let ServiceProvider {
-            parent,
-            dependencies,
-        } = self;
+impl<Parent, Conts: HList> ServiceProvider<Parent, Conts> {
+    fn add<Cont: Container>(self, data: Cont::Data) -> ServiceProvider<Parent, HCons<Cont, Conts>> {
```

Comment:
I'm most unsure about making this method private, since it seems like you are trying to allow custom-defined containers to be added to ServiceProvider. The method had an underscore prefix which tells me as a dev I'm not supposed to use this method in normal scenarios i.e the method should be treated as private. I just solidified that concept but can revert if we do want to make it public. I'll just remove the `_` if thats the case.

Reply:
The user might want to add their own container, so we need to give them a way to do this. I agree that the `_add()` function is a bad way, so we need to find another (maybe not in this PR).

Reply:
`add()` might be a good enough name, or maybe even `add_container()`, but I'll revert my change regardless.